### ISSUE TITLE
clang-tidy: cppcoreguidelines-owning-memory doesn't flag delete non-owner members

### ISSUE THREAD

**Issue by qqiangwu** [llvm/llvm-project#79554]
```cpp
struct Test
{
    int* p = nullptr;

    Test(const Test&) = delete;
    Test& operator=(const Test&) = delete;
    Test(Test&&) = delete;
    Test& operator=(Test&&) = delete;

    ~Test()
    {
        delete p;
    }
};
```

the above code is not flagged, but I hope `delete p;` be flagged.

**Comment by vbvictor** [comment]
https://godbolt.org/z/xo5dYaEEj
